rest/rest_collection.cc

```diff
@@ -752,6 +768,7 @@ getState() const
     case State::_executing:
         return L"executing";
     default:
+        ExcAssert(!"update the BackgroundTaskBase::getState");
```

Comment:
Can you add an extra comment to ensure that we know what to do if this triggers?